File: MyClub/MyClub.Model/Requests/PaymentRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MyClub.Model.Requests
{
    public class PaymentRequest
    {
        [Required(ErrorMessage = "Metoda plaćanja je obavezna")]
        [RegularExpression("^(Stripe|PayPal)$", ErrorMessage = "Metoda plaćanja mora biti 'Stripe' ili 'PayPal'")]
        public string Type { get; set; }

        [Required(ErrorMessage = "Iznos je obavezan")]
        [Range(0.01, double.MaxValue, ErrorMessage = "Iznos mora biti veći od 0")]
        public decimal Amount { get; set; }
        public string? PaymentMethod { get; set; }

        public string? ReturnUrl { get; set; } // For PayPal return URL
        public string? CancelUrl { get; set; } // For PayPal cancel URL
    }
} 

File: MyClub/UI/myclub_mobile/lib/models/requests/payment_request.dart
/// Abstract base class for payment requests
abstract class PaymentRequest {
  final String type; // "Stripe" or "PayPal"
  final double amount;
  final String? paymentMethod;
  final String? returnUrl; // For PayPal return URL
  final String? cancelUrl; // For PayPal cancel URL

  PaymentRequest({
    required this.type,
    required this.amount,
    this.paymentMethod,
    this.returnUrl,
    this.cancelUrl,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'type': type,
      'amount': amount,
    };
    
    if (paymentMethod != null) {
      json['paymentMethod'] = paymentMethod!;
    }
    if (returnUrl != null) {
      json['returnUrl'] = returnUrl!;
    }
    if (cancelUrl != null) {
      json['cancelUrl'] = cancelUrl!;
    }
    
    return json;
  }
}
